### PR TITLE
🎉 New Source: Google Workspace Admin audit logs

### LINKED ISSUE #2456: Google Audit Logs
## Tell us about the new integration you’d like to have
I'd like to pull my Google Audit log data from the [Admin Activity Reports API](https://developers.google.com/admin-sdk/reports/v1/guides/manage-audit-admin) via Airbyte.

There are a number of reports that can be pulled: admin, drive, logins, mobile, oauth tokens. I would like to pull the following streams for the moment:
* admin
* logins

## Describe the context around this new integration
Airbyte is looking to cover the security, logging, and auditing category, and this is an important API to cover.   

## Describe the alternative you are considering or using
[SnowAlert](https://github.com/snowflakedb/SnowAlert/blob/master/src/connectors/gsuite_logs.py) has a plugin for this that we can use as a learning point

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## What
Created new source for pulling audit logs #2456  via Google Workspace > Admin SDK > Reports API. It implementing only full refresh for now. Incremental sync will be released in separate PR.

Supported streams:

- Admin
- Drive
- Logins
- Mobile
- OAuth Tokens

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

### CODE REVIEW

File: docs/integrations/sources/google-workspace-admin-reports.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+| `array` | `array` |  |
+| `object` | `object` |  |
+
+### Features
+
+| Feature | Supported?\(Yes/No\) | Notes |
+| :--- | :--- | :--- |
+| Full Refresh Sync | Yes |  |
+| Incremental Sync | No |  |
+| Replicate Incremental Deletes | Coming soon |  |
```
Comment (sherifnada) [inline docs/integrations/sources/google-workspace-admin-reports.md:32]:
```suggestion
```

**Review by sherifnada — APPROVED** [review]
Great turnaround! LGTM assuming tests pass.

File: airbyte-integrations/connectors/source-google-workspace-admin-reports/source_google_workspace_admin_reports/utils.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+
+def rate_limit_handling(error):
+    retried_cases = [
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-workspace-admin-reports/source_google_workspace_admin_reports/utils.py:27]: should this contain 429?

Comment (vitaliizazmic) [reply airbyte-integrations/connectors/source-google-workspace-admin-reports/source_google_workspace_admin_reports/utils.py:27]: According to documentation If limits are exceeded, API returns an HTTP 503 status code.

Comment (sherifnada) [reply airbyte-integrations/connectors/source-google-workspace-admin-reports/source_google_workspace_admin_reports/utils.py:27]: ok 👍🏼

**Comment by sherifnada** [comment]
Lookback window may help solve the massive number of records problem

File: airbyte-integrations/connectors/source-google-workspace-admin-reports/source_google_workspace_admin_reports/spec.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      },
+      "email": {
+        "type": "string",
+        "description": "The email of the user, which has permissions to access the Google Workspace Admin APIs."
+      },
+      "lookback": {
+        "type": "integer",
+        "minimum": 0,
+        "maximum": 180,
+        "description": "Sets the range of time shown in the report. Reports API allow to set timespan less or equal 180 days. In another case it will restrict it."
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-workspace-admin-reports/source_google_workspace_admin_reports/spec.json:23]:
```suggestion
        "description": "Sets the range of time shown in the report. Reports API allows from up to 180 days ago. "
```

**Review by sherifnada — APPROVED** [review]

**Comment by vitaliizazmic** [comment]
/publish connector=connectors/source-google-workspace-admin-reports

**Pull request merged**